### ISSUE TITLE
[BUG] Developer Guide does not share information on how to build an extension. 

### ISSUE THREAD

**Issue by minalsha** [opensearch-project/opensearch-sdk-java#603]
### What is the bug?
Developer Guide: https://github.com/opensearch-project/opensearch-sdk-java/blob/main/DEVELOPER_GUIDE.md#submitting-changes does not share information on how to build an extension. Developer guide should be a one-stop shop which should cover the following:
1. How to build an extension from scratch? (include a sample Example)
3. How to run an extension?
4. How to migrate an existing plugin to an extension?

**Comment by owaiskazi19** [comment]
@dbwiddis can we close this one out?

**Comment by dbwiddis** [comment]
> @dbwiddis can we close this one out?

Nope, I only addressed a portion of the 1st point (and 3 is also a work in progress) but part 2 hasn't yet been worked on.  @saratvemulapalli will probably address it.
